include/deal.II/base/polynomial.h

```diff
@@ -585,6 +586,119 @@ namespace Polynomials
+   * first polynomial $p_0$ to 1 in $x=0$ and has a both a zero value and zero
+   * derivative in $x=1$. Likewise, the last polynomial $p_n$ evaluates to 1
+   * in $x=1$ but has zero value and zero derivative in $x=0$. The second
+   * polynomial $p_1$ and the second to last polynomial $p_{n-1}$ represent
+   * the derivative degree of freedom in $x=0$ and $x=1$, respectively. As
+   * such, they are zero in both the end points $x=0, x=1$ and have zero
+   * derivative in the opposite ends, $p_1'(1)=0$ and $p_{n-1}'(0)=0$. As
+   * opposed to the original Hermite polynomials, $p_0$ does not have zero
+   * derivative in $x=0$. The additional degree of freedom is used to make
+   * $p_0$ and $p_1$ orthogonal, which results in a root at $2/7$ for
+   * $n=3$. Furthermore, the extension of these polynomials to higher degrees
+   * $n>3$ is constructed by adding additional nodes inside the unit interval,
+   * again ensuring better conditioning. The nodes are computed as the roots
+   * of the Jacobi polynomials for $\alpha=\beta=2$ which are orthogonal
+   * against the generating function $x^2(1-x)^2$ of the Hermite
+   * functions. For example at $n=4$, all of $p_0, p_1, p_3, p_4$ get an
+   * additional root at $x=0.5$ through the factor $(x-0.5)/x_0$.
+
+   * These two relaxations improve the condition number of the mass matrix
+   * (i.e., interpolation) significantly, as can be seen from the following
+   * table:
+   *
+   * <table align="center" border="1">
+   *   <tr>
+   *    <th>&nbsp;</th>
+   *    <th colspan="2">Condition number mass matrix</th>
+   *   </tr>
+   *   <tr>
+   *    <th>degree</th>
+   *    <th>HermiteInterpolation</th>
+   *    <th>HermiteLikeInterpolation</th>
+   *   </tr>
+   *   <tr>
+   *    <th>n=3</th>
+   *    <th>1057</th>
+   *    <th>21.40</th>
+   *   </tr>
+   *   <tr>
+   *    <th>n=4</th>
+   *    <th>6580</th>
+   *    <th>15.52</th>
+   *   </tr>
+   *   <tr>
+   *    <th>n=5</th>
+   *    <th>1.875e+04</th>
+   *    <th>18.52</th>
+   *   </tr>
+   *   <tr>
+   *    <th>n=6</th>
+   *    <th>6.033e+04</th>
+   *    <th>19.42</th>
+   *   </tr>
+   *   <tr>
+   *    <th>n=10</th>
+   *    <th>9.756e+05</th>
+   *    <th>27.85</th>
+   *   </tr>
+   *   <tr>
+   *    <th>n=15</th>
+   *    <th>9.431e+06</th>
+   *    <th>40.48</th>
+   *   </tr>
+   *   <tr>
+   *    <th>n=25</th>
+   *    <th>2.220e+08</th>
+   *    <th>68.30</th>
+   *   </tr>
+   *   <tr>
+   *    <th>n=35</th>
+   *    <th>2.109e+09</th>
+   *    <th>98.06</th>
+   *   </tr>
+   * </table>
+   *
+   * This polynomial inherits the advantageous property of Hermite polynomials
+   * where only two functions have value and/or derivative nonzero on a face
+   * but gives better condition numbers of interpolation, which improves the
+   * performance of some iterative schemes like conjugate gradients with
+   * point-Jacobi.
+   *
+   * @note This class requires LAPACK support.
```

Comment:
This is one point I wanted to discuss: This class (and subsequently `FE_DGQHermite`) needs LAPACK support for degree larger than 4 because we need to solve a symmetric eigenvalue problem for the zeros of the Jacobi polynomial that are underlying the stable interpolation. I think that any sane installation will provide LAPACK but it's still something we should have in mind. The alternative would be to hardcode the points up to degree say 10 or 12 (it's only a handful of values after all) and require LAPACK beyond that. Any opinions? (My actual preference would be to make LAPACK a required dependency of deal.II - as many other scientific packages do.)

Reply:
`FE_DGQHermite` is not a standard choice (yet). Hence, I don't really see a problem in requiring `LAPACK` for this class. For more functionality we just need to build with more dependencies. That is true for many more classes.

The question whether we want to make `LAPACK` a required dependency can certainly be split into another issue. Cleaning up afterwards should be rather simple.

Reply:
> The question whether we want to make LAPACK a required dependency can certainly be split into another issue. Cleaning up afterwards should be rather simple.

Of course, we should not discuss that here, I just brought it up to see if there are any reactions 😄

Reply:
> My actual preference would be to make LAPACK a required dependency of deal.II - as many other scientific packages do.

👍